src/Standard.Licensing.Tests/LicenseValidationTests.cs

```diff
@@ -226,28 +219,31 @@ public void Do_Not_Crash_On_Invalid_Data()
         [Test]
         public void Test_ValidationChainBuilder_ValidationFailure_List()
         {
-            var keyGenerator = Standard.Licensing.Security.Cryptography.KeyGenerator.Create();
-            var keyPair = keyGenerator.GenerateKeyPair();
-            var publicKey = keyPair.ToPublicKeyString();
+            KeyGenerator? keyGenerator = KeyGenerator.Create();
+            KeyPair? keyPair = keyGenerator.GenerateKeyPair();
+            string? publicKey = keyPair.ToPublicKeyString();
 
-            var invalidLicense = @"<License>
+            const string invalidLicense = @"<License>
   <Signature>WFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFg=</Signature>
 </License>";
 
-            var licenseToVerify = License.Load(invalidLicense);
+            License? licenseToVerify = License.Load(invalidLicense);
 
-            var validationFailures = licenseToVerify
-                .Validate()
-                .Signature(publicKey)
-                .AssertValidLicense();
+            IEnumerable<IValidationFailure>? validationFailures = licenseToVerify
+                                                                  .Validate()
+                                                                  .Signature(publicKey)
+                                                                  .AssertValidLicense();
 
-            var count = 0;
-            foreach (var v in validationFailures)
-                count++;
+            int count = 0;
+            IEnumerable<IValidationFailure> enumerable = validationFailures.ToList();
+            foreach (IValidationFailure? v in enumerable)
+            {
+                count++;
+            }
 
             Assert.That(count, Is.EqualTo(1));
-            Assert.That(validationFailures.ToArray().Length, Is.EqualTo(1));
-            Assert.That(validationFailures.ToArray().Length, Is.EqualTo(1));
+            Assert.That(enumerable.ToArray().Length, Is.EqualTo(1));
+            Assert.That(enumerable.ToArray().Length, Is.EqualTo(1));
```

Comment:
To be honest, I am not following what the original test idea was, but the behavior here has been changed. In the old test the validationFailures enumerable is being enumerated 3 times. With the code change it is being done once in the ToList() call and then everything else is being done on the resulting list. I would have to look deeper at the test to understand whether the old test was intentionally performing multiple enumerations to ensure that it was possible. In which case then the change to calling ToList() and removing the multiple enumerations is changing the test behavior.